### ISSUE TITLE
Unable to compile

### ISSUE THREAD

**Issue by RicoloveFeng** [juliannagele/ebso#1]
Hi! Recently, I read your paper and was interested in how this tool works.

However, when I was trying to compile ebso, it always failed and reported somthing like:

```
[ERROR] The compilation of ebso.2.1 failed at "dune build -p ebso -j 15".

#=== ERROR while compiling ebso.2.1 ===========================================#
# context     2.1.0 | linux/x86_64 | ocaml-base-compiler.4.08.0 | pinned(git+file:///mnt/d/gitrepo/ebso#HEAD#3862ed57)
# path        ~/.opam/4.08.0/.opam-switch/build/ebso.2.1
# command     ~/.opam/4.08.0/bin/dune build -p ebso -j 15
# exit-code   1
# env-file    ~/.opam/log/ebso-29318-07a107.env
# output-file ~/.opam/log/ebso-29318-07a107.out
### output ###
# [...]
# (cd _build/default && /home/ricolove/.opam/4.08.0/bin/ocamlc.opt -w -40 -g -bin-annot -I lib/.ebso.objs/byte -I /home/ricolove/.opam/4.08.0/lib/base -I /home/ricolove/.opam/4.08.0/lib/base/base_internalhash_types -I /home/ricolove/.opam/4.08.0/lib/base/caml -I /home/ricolove/.opam/4.08.0/lib/base/md5 -I /home/ricolove/.opam/4.08.0/lib/base/shadow_stdlib -I /home/ricolove/.opam/4.08.0/lib/base[...]
# File "lib/program.ml", line 166, characters 48-70:
# 166 |           let is = List.filter cis ~f:(fun i -> Instruction.gas_cost i = Gas_cost.of_int (g - pg)) in
#                                                       ^^^^^^^^^^^^^^^^^^^^^^
# Error: This expression has type Ebso.Gas_cost.t
#        but an expression was expected of type Core_kernel__Int.t = int
# (cd _build/default && /home/ricolove/.opam/4.08.0/bin/ocamlopt.opt -w -40 -g -I lib/.ebso.objs/byte -I lib/.ebso.objs/native -I /home/ricolove/.opam/4.08.0/lib/base -I /home/ricolove/.opam/4.08.0/lib/base/base_internalhash_types -I /home/ricolove/.opam/4.08.0/lib/base/caml -I /home/ricolove/.opam/4.08.0/lib/base/md5 -I /home/ricolove/.opam/4.08.0/lib/base/shadow_stdlib -I /home/ricolove/.opam[...]
# File "lib/word.ml", line 76, characters 4-26:
# 76 |     Z.of_string (to_dec w) < max_repr
#          ^^^^^^^^^^^^^^^^^^^^^^
# Error: This expression has type Z.t but an expression was expected of type
#          Core_kernel__Int.t = int
```  

I'm using WSL2 and has tried ocaml from 4.07 to the latest version.

The only case that the tool being successfully compiled was with ocaml 4.07, but the tool itself cannot run:

```
ebso: symbol lookup error: ebso: undefined symbol: Z3_get_parser_error
```

So how can I compile ebso and run it? Should I use a virtual machine, or use an older version of ocaml?

Thank you!

**Comment by juliannagele** [comment]
Hi and thanks for trying to give ebso a spin!

Version 2.1 should compile ok with OCaml 4.07.0, for anything newer the current develop branch should do it. I'm afraid I can't reproduce that symbol lookup error (I don't have access to a Windows machine so not sure if WSL is the problem), looks like something went wrong while linking though.

**Comment by RicoloveFeng** [comment]
Thank you for your information, but sadly it didn't work.

Ocaml 4.07.0 with WSL did make the compilation done. But the problem is, when I run `ebso -help`, the `Z3_get_parser_error` message always pops up. I don't know whether it is a compiling bug, or a dependency problem.

I switched the environment to Ubuntu 20.04 and installed ocaml 4.07.0, and it still popped the same symbol lookup error.

I have tried the master branch and the develop branch, but none of them made ebso work.

Also, I further tried ocaml 4.06.0, which just returned the same error.

So I wonder whether there are some dependencies that should be installed with `apt` instead of `opam`? (Opam did tell me to install `libgmp` and `python2.7` because I was using a newly installed Ubuntu, I don't know if that was enough.)
